### PR TITLE
Use volatile to fix serialize issue

### PR DESCRIPTION
**Pull request by ashiquzzaman33**
-Ref: https://github.com/appscode-ci/appscode-login-plugin/issues/12

### CODE REVIEW

**Comment by ashiquzzaman33** [comment]
volatile is not applicable for final field.
So Cache are remain unchanged.

File: src/main/java/com/appscode/ci/plugins/authn/AppsCodeSecurityRealm.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
   private static final Cache<UserId, String> emailCache = CacheBuilder.newBuilder()
```
Comment (TamalSaha) [inline src/main/java/com/appscode/ci/plugins/authn/AppsCodeSecurityRealm.java]: Cache should be also volatile.

File: src/main/java/com/appscode/ci/plugins/authn/AppsCodeSecurityRealm.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
...
       .expireAfterAccess(7, TimeUnit.DAYS).build();
   private static Auth currentAuth;
```
Comment (TamalSaha) [inline src/main/java/com/appscode/ci/plugins/authn/AppsCodeSecurityRealm.java]: CurrentAuth should be also volatile.

File: src/main/java/com/appscode/ci/plugins/authn/AppsCodeSecurityRealm.java (hunk 1 above)
Comment (ashiquzzaman33) [inline src/main/java/com/appscode/ci/plugins/authn/AppsCodeSecurityRealm.java]: Volatile is illegal with final modifier

**Pull request merged**